app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,31 @@
+class CommentsController < ApplicationController
+
+  def index
+    @comments = Comment.all
+  end
+
+  def show
+    @comment = Comment.find(params[:id])
+  end
+
+  def new
+    @comment = Comment.new
+  end
+
+  def create
+    @comment = Comment.create(comment_params)
```

Comment:
Creating a comment this way doesn't associate it with a post. See the solution for adding songs to artists in tunr
```rb
  # create
  def create
    @artist = Artist.find(params[:artist_id])
    @song = Song.create!(song_params.merge(artist: @artist))
    redirect_to artist_song_path(@artist, @song)
  end
```